core/domain/voiceover_domain.py

```diff
@@ -316,3 +317,100 @@ def create_empty(
             language_accent_code=language_accent_code,
             voiceovers_mapping={},
             automated_voiceovers_audio_offsets_msecs={})
+
+
+class ExplorationVoiceArtistsLink:
+    """Domain object for linking Exploration with voice artists."""
+
+    def __init__(
+            self,
+            exp_id: str,
+            content_id_to_voiceovers_mapping: Dict[
+                str, Dict[str, Tuple[str, state_domain.VoiceoverDict]]
+            ]
+    ) -> None:
+        """Construct a ExplorationVoiceArtistsLink domain object.
+
+        Args: 
+            exp_id: str. ID of exploration.
+            content_id_to_voiceovers_mapping: Dict. A dictionary with content
+                IDs as keys and nested dicts as values. Each nested dict
+                contains language codes as keys and a 2-tuple as values.
+                The 2-tuple contains voice artist ID as the first element 
+                and VoiceoverDict as the second element.
+        """
+        self.exp_id = exp_id
+        self.content_id_to_voiceovers_mapping = content_id_to_voiceovers_mapping
+
+    def validate(self) -> None:
+        """Validates attributes of ExplorationVoiceArtistsLink."""
+
+        if not isinstance(self.exp_id, str):
+            raise utils.ValidationError(
+                'Expected exploration id to be string, recieved: %s.'
+                % self.exp_id
+            )
+        if self.exp_id == '':
+            raise utils.ValidationError(
+                'Exploration ID should not be empty.'
+            )
+        if not isinstance(self.content_id_to_voiceovers_mapping, dict):
+            raise utils.ValidationError(
+                'Expected content_id_to_voiceovers_mapping to be Dict'
+                'But recieved: %s' % self.content_id_to_voiceovers_mapping
+            )
+        # Validating content within content_id_to_voiceovers_mapping dict.
+        for content_id, nested_dict in (
+        self.content_id_to_voiceovers_mapping.items()):
```

Comment:
Fix indent.

Reply:
This is still not fixed, you need to indent when you break after a bracket.
```suggestion
        for content_id, nested_dict in (
            self.content_id_to_voiceovers_mapping.items()):
```